sdk/core/core/inc/az_context.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file az_context.h
+ *
+ * @brief az_context for cancelling execution
+ */
+
+#ifndef _az_CONTEXT_H
+#define _az_CONTEXT_H
+
+#include <az_result.h>
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <_az_cfg_prefix.h>
+
+typedef struct az_context az_context;
```

Comment:
Can't you define it as `typedef struct { ... } az_context`?

Reply:
no, because the definition of az_context includes az_context inside of it